conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -487,12 +491,50 @@ private CodeBlock generateParameterCodeBlock(Stream<ArgumentDefinition> params,
                                         arg.getArgName().get(),
                                         createConstructorForTypeWithReference(arg.getType(), rawVarName,
                                                 typeDefinitions, typeMapper)
-                                )
+                                ),
+                                generateStoreParamInExchange(arg, paramTypeVisitor)
                         );
                     }
                 }).collect(Collectors.toList()));
     }
 
+    private Optional<String> getStoreParamMethodName(
+            ArgumentDefinition arg,
+            ParameterType.Visitor<Boolean> visitor) {
+        boolean isSafe = arg.getMarkers().stream().anyMatch(type ->
+                type.accept(
+                        new DefaultTypeVisitor<Boolean>() {
+                            @Override
+                            public Boolean visitExternal(ExternalReference value) {
+                                return value.getExternalReference().getName().equals("Safe")
+                                        && value.getExternalReference().getPackage().equals("com.palantir.redaction");
```

Comment:
isn't the right package `com.palantir.logsafe`?

Reply:
It appears it is not, else the generator tests would not pass.

Reply:
in the undertow case we now need to encode the class that would *actually* be used, not the one used in isolation in tests. https://github.com/palantir/safe-logging/blob/develop/safe-logging/src/main/java/com/palantir/logsafe/Safe.java

Reply:
Ah that makes sense, I thought Safe was an hardcoded/pre-imported marker and that it was always com.palantir.redaction.
There are a few alternatives:
- filter on getName == Safe exclusively (my preferred option, I don't think we will have many cases of conflict on the marker for Safe)
- filter on union of com.palantir.redaction.Safe and com.palantir.logsafe.Safe (pretty sad)
- rewrite the tests so that they depend on logsafe

What do you prefer?